lib/ansible/cli/galaxy.py

```diff
@@ -146,6 +146,23 @@ def _get_collection_widths(collections):
     return fqcn_length, version_length
 
 
+def validate_signature_count(value):
+    value = str(value)
+
+    if value.startswith('+'):
```

Comment:
This could be more robust with something like `^(?P<strict>\+)?(?:(?P<count>\d+)|(?P<all>all))$` - then you can just ask for the various named bits on a match with full assurance that you've got something valid, and just make the error message on a non-match generic like `(bla) is not a valid signature count value`. As-is, there are several cases that would just blow up with a generic `ValueError` from `int`.

Reply:
Yeah, that's much better. Also fixes floats, which shouldn't be valid!